src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -186,108 +213,169 @@ internal void UpdateSelectedItem(PackageItemListViewModel selectedItem)
             if (selectedItem != null)
             {
                 // select the the previously selected item if it still exists.
-                selectedItem = PackageItems
+                selectedItem = PackageItemsFiltered
                     .FirstOrDefault(item => item.Id.Equals(selectedItem.Id, StringComparison.OrdinalIgnoreCase));
             }
 
             // select the first item if none was selected before
-            _list.SelectedItem = selectedItem ?? PackageItems.FirstOrDefault();
+            _list.SelectedItem = selectedItem ?? PackageItemsFiltered.FirstOrDefault();
         }
 
-        private void LoadItems(PackageItemListViewModel selectedPackageItem, CancellationToken token)
+        private async Task LoadItems(PackageItemListViewModel selectedPackageItem, CancellationToken token)
```

Comment:
Now that it is async, can we rename it to LoadItemsAsync ?

```suggestion
        private async Task LoadItemsAsync(PackageItemListViewModel selectedPackageItem, CancellationToken token)
```